sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ExprUtils.scala

```diff
@@ -40,8 +40,8 @@ object ExprUtils extends EvalHelper with QueryErrorsBase {
         case s: UTF8String if s != null =>
           val dataType = DataType.parseTypeWithFallback(
             s.toString,
-            DataType.fromDDL,
-            DataType.fromJson)
+            DataType.fromJson,
+            DataType.fromDDL)
```

Comment:
Won't your fix work without changing the order. When `DataType.fromDDL` fails, and it should fail, the next one `DataType.fromJson` shouldn't fail. Correct me if I am wrong.

Reply:
You are right, but won't it make sense to do it from json first? Pyspark also does this using `toJson` https://github.com/apache/spark/blob/91c01dbad451254e0e3f5efc3e33b7f5d612b5fa/python/pyspark/sql/functions/builtin.py#L18299

Reply:
Comparing to python code, `ExprUtils.evalTypeExpr` is used from many more places, and you might acidentally change behaviour there.

Reply:
okay, I reverted it back